src/variables/components/VariableContextMenu.tsx

```diff
@@ -19,29 +29,55 @@ export default class VariableContextMenu extends PureComponent<Props> {
   public render() {
     const {variable, onExport, onRename, onDelete} = this.props
 
+    const settingsRef: RefObject<HTMLButtonElement> = createRef()
     return (
-      <Context>
-        <Context.Menu icon={IconFont.CogThick}>
-          <Context.Item label="Export" action={onExport} />
-          <Context.Item
-            label="Rename"
-            action={onRename}
-            testID="context-rename-variable"
-          />
-        </Context.Menu>
-        <Context.Menu
-          icon={IconFont.Trash}
-          color={ComponentColor.Danger}
-          testID="context-delete-menu"
-        >
-          <Context.Item
-            label="Delete"
-            action={onDelete}
-            value={variable}
-            testID={`context-delete-variable ${variable.name}`}
-          />
-        </Context.Menu>
-      </Context>
+      <FlexBox margin={ComponentSize.ExtraSmall}>
+        <ConfirmationButton
+          color={ComponentColor.Colorless}
+          icon={IconFont.Trash_New}
+          shape={ButtonShape.Square}
+          size={ComponentSize.ExtraSmall}
+          confirmationLabel="Yes, Delete this Variable"
+          onConfirm={() => {
+            onDelete(variable)
```

Comment:
same thing here - pull this function out into a function that isn't created on render. since it only accesses props, it doesn't need to be re-made every render cycle